src/renderer/components/dock/pod-log-list.tsx

```diff
@@ -0,0 +1,224 @@
+    } else {
+      this.isJumpButtonVisible = true;
+    }
+  };
+
+  /**
+   * Checks if last log line considered visible to user, setting its observable
+   * @param props Scrolling props from virtual list core
+   */
+  @action
+  setLastLineVisibility = (props: ListOnScrollProps) => {
+    const { scrollHeight, clientHeight } = this.virtualListDiv.current;
+    const { scrollOffset, scrollDirection } = props;
+    if (scrollDirection == "backward") {
+      this.isLastLineVisible = false;
+    } else {
+      if (clientHeight + scrollOffset === scrollHeight) {
+        this.isLastLineVisible = true;
+      }
+    }
+  };
+
+  /**
+   * Check if user scrolled to top and new logs should be loaded
+   * @param props Scrolling props from virtual list core
+   */
+  checkLoadIntent = (props: ListOnScrollProps) => {
+    const { scrollOffset } = props;
+    if (scrollOffset === 0) {
+      this.props.load();
+    }
+  };
+
+  @action
+  scrollToBottom = () => {
+    if (!this.virtualListDiv.current) return;
+    this.isJumpButtonVisible = false;
+    this.virtualListDiv.current.scrollTop = this.virtualListDiv.current.scrollHeight;
+  };
+
+  scrollToItem = (index: number, align: Align) => {
+    this.virtualListRef.current.scrollToItem(index, align);
+  };
+
+  onScroll = debounce((props: ListOnScrollProps) => {
+    if (!this.virtualListDiv.current) return;
+    this.setButtonVisibility(props);
+    this.setLastLineVisibility(props);
+    this.checkLoadIntent(props);
+  }, 700); // Increasing performance and giving some time for virtual list to settle down
+
+  /**
+   * A function is called by VirtualList for rendering each of the row
+   * @param rowIndex index of the log element in logs array
+   * @returns A react element with a row itself
+   */
+  getLogRow = (rowIndex: number) => {
+    const { searchQuery, isActiveOverlay } = searchStore;
+    const item = this.props.logs[rowIndex];
+    const contents: React.ReactElement[] = [];
+    const ansiToHtml = (ansi: string) => DOMPurify.sanitize(colorConverter.ansi_to_html(ansi));
+    if (searchQuery) { // If search is enabled, replace keyword with backgrounded <span>
+      // Case-insensitive search (lowercasing query and keywords in line)
+      const regex = new RegExp(searchStore.escapeRegex(searchQuery), "gi");
+      const matches = item.matchAll(regex);
+      const modified = item.replace(regex, match => match.toLowerCase());
+      // Splitting text line by keyword
+      const pieces = modified.split(searchQuery.toLowerCase());
+      pieces.forEach((piece, index) => {
+        const active = isActiveOverlay(rowIndex, index);
+        const lastItem = index === pieces.length - 1;
+        const overlayValue = matches.next().value;
+        const overlay = !lastItem ?
+          <span
+            className={cssNames("overlay", { active })}
+            dangerouslySetInnerHTML={{ __html: ansiToHtml(overlayValue) }}
+          /> :
+          null;
```

Comment:
This format is rather confusing to read.
```suggestion
        const overlay = !lastItem
          ? <span
            className={cssNames("overlay", { active })}
            dangerouslySetInnerHTML={{ __html: ansiToHtml(overlayValue) }}
          />
          : null;
```